lib/views/map/map.dart

```diff
@@ -1,44 +1,235 @@
+import 'dart:async';
 import 'package:flutter/material.dart';
 import 'package:flutter_map/flutter_map.dart';
-import 'package:flutter_svg/flutter_svg.dart';
+import 'package:flutter_map_location_marker/flutter_map_location_marker.dart';
 import 'package:latlong2/latlong.dart';
+import 'package:geolocator/geolocator.dart';
 import 'package:wildgids/config/theme/asset_icons.dart';
+import 'package:wildgids/services/animal.dart';
+import 'package:wildgids/services/tracking.dart';
+import 'package:wildlife_api_connection/models/animal_tracking.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:wildlife_api_connection/models/species.dart';
+
+class MapView extends StatefulWidget {
+  final AnimalService animalService;
+  final TrackingService trackingService;
+
+  const MapView({
+    super.key,
+    required this.animalService,
+    required this.trackingService,
+  });
+
+  @override
+  MapViewState createState() => MapViewState();
+}
+
+class MapViewState extends State<MapView> {
+  // Set global properties
+  late Timer _animalTimer;
+  late Timer _trackingTimer;
+  List<Marker> _markers = [];
+  LatLng? _initialPosition;
+  final MapController _mapController = MapController();
+
+  // On initial state, set the initial location and start fetching markers with a timer
+  @override
+  void initState() {
+    super.initState();
+    _setInitialLocation();
+    _startTimers();
+
+    // Fetch markers and send tracking on initial load
+    _sendTracking();
+    _fetchMarkers();
+  }
+
+  @override
+  void dispose() {
+    _animalTimer.cancel();
+    _trackingTimer.cancel();
+    super.dispose();
+  }
+
+  // Set initial location for map bounds and camera center
+  void _setInitialLocation() async {
+    Position position = await _determinePosition();
+
+    setState(() {
+      _initialPosition = LatLng(position.latitude, position.longitude);
+    });
+  }
+
+  // Start timer to retrieve/upload location and fetch markers
+  void _startTimers() async {
+    // Set timer to retrieve location every 10 seconds
+    _animalTimer = Timer.periodic(const Duration(seconds: 10), (timer) async {
+      _fetchMarkers();
+    });
+
+    _trackingTimer = Timer.periodic(const Duration(minutes: 1), (timer) async {
+      _sendTracking();
+    });
+  }
+
+  // Get the current location of the device
+  Future<Position> _determinePosition() async {
+    bool serviceEnabled;
+    LocationPermission permission;
+
+    serviceEnabled = await Geolocator.isLocationServiceEnabled();
+    if (!serviceEnabled) {
+      return Future.error('Location services are disabled.');
+    }
+
+    // Check location permissions
```

Comment:
I make LocationManager class to handle fetching the permission. Maybe refactor your code before to use that class before merging this into development.

Reply:
This LocationManager wasn't available on my branch at the time of creation, I will do so once your feature branch is merged and conflits are resolved.